lib/install/management/create.go

```diff
@@ -109,10 +111,84 @@ func (d *Dispatcher) createPool(conf *config.VirtualContainerHostConfigSpec, set
 	return nil
 }
 
-func (d *Dispatcher) startAppliance(conf *config.VirtualContainerHostConfigSpec) error {
+// relocateAppliance invokes the host placement library to find a suitable host (based on host
+// metrics) to relocate (migrate) the VCH to. If the VCH's current host is suitable, the VCH
+// is not relocated.
+func (d *Dispatcher) relocateAppliance() error {
 	defer trace.End(trace.Begin("", d.op))
+	var err error
+
+	// TODO(anchal): Use ranked host policy when #7459 is merged.
+	// provider := performance.NewHostMetricsProvider(d.session)
+	// rankedPolicy := placement.NewRankedHostPolicy()
+
+	randomPolicy := placement.NewRandomHostPolicy()
+	if randomPolicy.CheckHost(d.op, d.appliance) {
+		// The current host of the appliance is suitable; no migration needed.
+		return nil
+	}
 
+	// TODO(anchal): Use this form when #7459 is merged.
+	// Collect a ranked slice of hosts and pick the first one to relocate the VCH VM to.
+	// Pass a nil slice of hosts to use all hosts in the cluster as candidate hosts.
+	// hosts, err := randomPolicy.RecommendHost(d.op, d.appliance, nil)
+	// if err != nil {
+	// 	msg := "Unable to obtain recommended host: %s"
+	// 	d.op.Warnf(msg, err)
+	// 	return errors.Errorf(msg, err)
+	// }
+	// if len(hosts) == 0 {
+	// 	msg := "No hosts returned by placement library, skipping relocation"
+	// 	d.op.Warnf(msg)
+	// 	return errors.New(msg)
+	// }
+	// hMoref := hosts[0].Reference()
+
+	host, err := randomPolicy.RecommendHost(d.op, d.appliance)
+	if err != nil {
+		msg := "Unable to obtain recommended host: %s"
+		d.op.Warnf(msg, err)
+		return errors.Errorf(msg, err)
+	}
+
+	hMoref := host.Reference()
```

Comment:
Is this really needed?  I don't think calling `host.Reference()` two times is going to be an issue and it would be more clear